salalib/segmmodules/segmtopological.cpp

```diff
@@ -0,0 +1,217 @@
+                    bool seenalready = (seen[connected_cursor] == 0xffffffff) ? false : true;
+                    float length = seglengths[connected_cursor];
+                    int axialref = axialrefs[connected_cursor];
+                    audittrail[connected_cursor] =
+                        TopoMetSegmentRef(connected_cursor, here.dir, here.dist + length, here.ref);
+                    seen[connected_cursor] = segdepth;
+                    if (options.radius == -1 || here.dist + length < options.radius) {
+                        // puts in a suitable bin ahead of us...
+                        open++;
+                        //
+                        if (axialrefs[here.ref] == axialref) {
+                            list[bin].push_back(connected_cursor);
+                        } else {
+                            list[(bin + 1) % 2].push_back(connected_cursor);
+                            seen[connected_cursor] =
+                                segdepth + 1; // this is so if another node is connected directly to this one but
+                                              // is found later it is still handled -- note it can result in the
+                                              // connected cursor being added twice
+                        }
+                    }
+                    // not sure why this is outside the radius restriction
+                    // (sel_only: with restricted selection set, not all lines will be labelled)
+                    // (seenalready: need to check that we're not doing this twice, given the seen can go twice)
+
+                    // Quick mod - TV
+                    if (!options.sel_only && connected_cursor > int(cursor) &&
+                        !seenalready) { // only one way paths, saves doing this twice
+                        int subcur = connected_cursor;
+                        while (subcur != -1) {
+                            // in this method of choice, start and end lines are included
+                            choicevals[subcur].choice += 1;
+                            choicevals[subcur].wchoice += (rootseglength * length);
+                            subcur = audittrail[subcur].previous;
+                        }
+                    }
+                }
+                iter++;
+            }
+        }
+        // also put in mean depth:
+        //
+        attributes.setValue(cursor, meandepthcol.c_str(), totalsegdepth / (total - 1));
+        attributes.setValue(cursor, totaldcol.c_str(), totalsegdepth);
+        attributes.setValue(cursor, wmeandepthcol.c_str(), wtotaldepth / (wtotal - rootseglength));
+        attributes.setValue(cursor, totalcol.c_str(), total);
+        attributes.setValue(cursor, wtotalcol.c_str(), wtotal);
+        //
+        if (comm) {
+            if (qtimer(atime, 500)) {
+                if (comm->IsCancelled()) {
+                    delete[] seen;
```

Comment:
Can we please, please get rid of these raw pointers while you're in here anyway? They make my toenails curl!